src/Lean/Data/Lsp/Capabilities.lean

```diff
@@ -33,7 +33,7 @@ structure TextDocumentClientCapabilities where
   deriving ToJson, FromJson
 
 structure ShowDocumentClientCapabilities where
-  support : Bool
+  support? : Option Bool := none
```

Comment:
Let's try to conform to the LSP spec. Per the spec, the `applyEdit` field should be optional so the other change is correct, but this field is not optional. Do you need it to be optional? `WindowClientCapabilities.showDocument` is already optional, so one can just omit the whole struct.